Sources/Services/Health/HealthService.swift

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+
+@available(macOS 15.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+internal struct HealthService: Grpc_Health_V1_HealthServiceProtocol {
+  private let state = HealthService.State()
+
+  func check(
+    request: ServerRequest.Single<Grpc_Health_V1_HealthCheckRequest>
+  ) async throws -> ServerResponse.Single<Grpc_Health_V1_HealthCheckResponse> {
+    let service = request.message.service
+
+    guard let status = self.state.currentStatus(ofService: service) else {
+      throw RPCError(code: .notFound, message: "Requested service unknown.")
```

Comment:
For consistency with `watch` below, should we throw here or return a `serviceUnknown` state?

Reply:
According to [the specification](https://github.com/grpc/grpc-swift/blob/main/Protos/upstream/grpc/health/v1/health.proto#L46) for `Check`, "If the requested service is unknown, the call will fail with status NOT_FOUND".